### PR TITLE
BCDA-7119: Add test coverage for main SSAS file

### PR DESCRIPTION
**Pull request by kyeah**
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-7119

## 🛠 Changes

- Adds testing for the main.go file

## ℹ️ Context for reviewers

This file is the main entrypoint for the program and all related scripts. In order to provide better test coverage over these lines of code, I had to restructure the methods a bit to rely less on global package variables and move most of the main() function into separate, smaller functions that could take in a Flags object.

I also split the start() function so that we could test the "CreateServers" functionality without actually starting web servers. I think we could _probably_ write a test that actually covers starting the web servers and hitting them with HTTP requests but I had some trouble figuring that out, so I left that alone for now. This should still cover a good amount of the file.

## ✅ Acceptance Validation

Unit tests pass.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

### CODE REVIEW

File: ssas/service/main/main.go [hunk 1]
```diff
@@ -52,86 +52,97 @@ import (
...
+	return flags
 }
 
-// We provide some simple commands for bootstrapping the system into place. Commands cannot be combined.
-func main() {
-	ssas.Logger.Info("Home of the System-to-System Authentication Service")
-
-	flag.Parse()
-	if doAddFixtureData {
+func handleFlags(flags Flags) {
```
Comment (laurenkrugen-navapbc) [inline ssas/service/main/main.go:122]: are these flags set from environment variables?

File: ssas/service/main/main.go [hunk 2]
```diff
@@ -52,86 +52,97 @@ import (
...
-var doResetSecret bool
-var doNewAdminSystem bool
-var doListIPs bool
-var doListExpCreds bool
-var doShowXData bool
-var doStart bool
-var clientID string
-var auth string
-var systemName string
+type Flags struct {
```
Comment (laurenkrugen-navapbc) [inline ssas/service/main/main.go:55]: this cleans up the tests and the flags nicely. Thanks for adding!

**Review by laurenkrugen-navapbc — APPROVED** [review]
LGTM! thanks for the context in the PR description.

File: ssas/service/main/main.go (hunk 1 above)
Comment (kyeah) [reply ssas/service/main/main.go:122]: They're passed in directly when the compiled go program is run, so something like `bcda --add-fixture-data` 😄

**Pull request merged**